### DISCUSSION TITLE
IP isn’t showing.

### DISCUSSION THREAD

**Discussion by AllyPallies** [cockpit-project/cockpit-machines#148] [Q&A]
Cockpit version: Latest
OS: Ubuntu
Page: VM page

looked everywhere I can’t figure out why my VM’s fail to show IP’s of the VM it’s using.

Steps to reproduce

1. Head to the VM page
2.click on any VM
3.head to the network interface I’m unable to see the VM’s IP for FreeBSD, Ubuntu and Windows I’m very unclear how to fix this or what package the VM’s need to install.

**Comment by KKoukiou** [comment]
@AllyPallies what's the output of:
```
$ sudo virsh domifaddr my-vm 
```
Does that contain an IP? Something like this:
```
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet0      52:54:00:b3:58:51    ipv4         192.168.122.225/24

```

If there is an IP listed there you should see it also in the VM details.
Make sure to replace 'my-vm' with your VM's name.

Also please note the exact cockpit-machines version, latest varies per distrobution and distro release so it's not very enlightening.

**Comment by AllyPallies** [comment]
@KKoukiou No that isnt showing anything honestly its like a package is missing or something odd is going on.
here is the output
willzor@mainserver:~$ sudo virsh domifaddr OpenSense
[sudo] password for willzor:
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------

willzor@mainserver:~$ sudo virsh list
 Id   Name        State
---------------------------
 1    pfsense     running
 2    FreeNAS     running
 4    OpenSense   running

willzor@mainserver:~$ sudo virsh domifaddr FreeNAS
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------

When i look at the VM display i get this

bridge | virtio | 52:54:00:69:7e:81 | Unknown | vmbridge1 | up
the IP should be where the "Unknown" part is. as i said im unsure whats going on but it sure feels like a package is missing or something super simple.

**Comment by KKoukiou** [comment]
@AllyPallies as you show above, it's not a problem of the UI, the cli can't detect any IP address either. Check this thread here, https://unix.stackexchange.com/a/549474/163281 

It has to do with how you have configured the NICs of your VM. I will move this issue to the discussion section, since it is something specific to your confuguration and I 'd like to keep issues list for generic bugs.

**Comment by AllyPallies** [comment]
That "thread" commands gave me zero results, commands didnt help in the slightist if you would like the close this thread please provide help as its either a "bug" or a VAILD question that needs to be answered.

**Reply by KKoukiou** [reply]
@AllyPallies I hope you understand that I am actually trying to help you. This is an issue specific to your VM NIC/Virtual network configuration however,  not a bug in cockpit-machines project itself (the project which you report this issue against). 

For this reason, I just moved this to the discussions section where support questions belong and I did not close it as you complain about.

So, the CLI reproducer that I made you run above is a proof that the IPs are not present by using the libvirt's CLI either. IOW the API that we are consuming in cockpit-machines does not have the IP information either.

So you can go forward as follows:

* Provide more information about you NIC configuration and Virtual Network configuration so that I have more information and try to help you
* Ask libvirt developers by  providing the CLI reproducer above, why you can't see yours IPs with virsh domifaddr

**Reply by AllyPallies** [reply]
Well i have a switch thats has 10 1GB connections into it going directly into the server thats hosting cockpit, then we have it setup as a Lag config for all ten to be into one. from there we have two Vlan's 1 and 10 Vlan 1 is for all the other devices on my network to talk to the server and Vlan 10 is my modem line coming in setup in a Lag config as well. so only the firewalls have access to Vlan10.

From there we use "host bridging" to give the VM's access to the that Vlan as you can see listed below. Interface type is Bridge to LAN.

bridge | virtio
vmbridge1 = Vlan1
vmbridge10 = Vlan10

**Reply by KKoukiou** [reply]
Ok in cockpit we only show IPs from those domains which take IP address from libvirt spawned dnsmasq (i.e. have <interface type='network'/>)

So for your NICs setup, we don't currently show IP addresses. I just noticed there is a duplicate issue here https://github.com/cockpit-project/cockpit-machines/issues/61 .

So, I will extend the functionality to cover other setups, so support fetching ip address with the other two ways, arp and guest agent.

Let's close this discussion, and keep track of the issue https://github.com/cockpit-project/cockpit-machines/issues/61 which has more informaton.

Note: You mentioned that arp did not work for you above, but guest agent should.

**Reply by AllyPallies** [reply]
Guest as well didnt give a positive response either. so i have no idea whats the cause or issue at this point.

**Reply by KKoukiou** [reply]
Check this https://github.com/cockpit-project/cockpit-machines/issues/61#issuecomment-704184607
There are basically three ways of getting IP address for a VM;
* we use the leases but that does not work for you and the other issue reported because it's only working for NICs using NAT networks
* I will implement the other two

**Reply by KKoukiou** [reply]
I sent a PR which will solve your issue: https://github.com/cockpit-project/cockpit-machines/pull/151
Please close this discussion and follow the original issue instead.
Thanks.

**Reply by AllyPallies** [reply]
@KKoukiou i don’t use IPV6 I’m using IPv4 looking over your “edit” I see it’s only IPv6 so that fix doesn’t work exactly for me.

**Reply by KKoukiou** [reply]
My PR constitutes of two commits, you 're interested in the first https://github.com/cockpit-project/cockpit-machines/pull/151/commits/8d5937c3c734415c4bad980fb136d178e8794781

The second adds ipv6 for the ones who use it.

In anycase, this issue is a duplicate of #61

**Reply by AllyPallies** [reply]
I can see you fixed this how would I update to this version to allow it to work? Do I have to manually edit the files or anyway to pull them without causing massive issues?

**Reply by KKoukiou** [reply]
It has to be merged, then released (probably in 244) and then get to ubuntu backports.

**Reply by KKoukiou** [reply]
The patch is merged https://github.com/cockpit-project/cockpit-machines/pull/151
Will be released in 244.
